### ISSUE TITLE
When using s3.read_parquet and passing columns to filter, encountering KeyError: "None of ['__index_level_0__'] are in the columns"

### ISSUE THREAD

**Issue by tasq-inc** [awslabs/aws-data-wrangler#408]
**Describe the bug**
When using `awswrangler.s3.read_parquet` and using the `columns` argument to filter on a parquet file that does not store the index (i.e., a file that exclude the `__index_level_0__` field), a KeyError is raised stating that said column isn't found.

I've tried both version 1.9.2 and 1.9.4.

**To Reproduce**
Given an example dataset stored in s3 like below that does not have a `__index_level_0__` field:
Example dataset:
time | Field1 | Field 2
-- | -- | --
2020-07-01 00:09:59+00:00 | 0.2 | 3
2020-07-01 00:29:59+00:00 | 5.1 | 2
2020-07-01 00:50:00+00:00 | 2.2 | 1
2020-07-01 01:10:00+00:00 | 3.6 | 4
2020-07-01 01:17:00+00:00 | 10.0 | 5

Retrieving the dataset with:
```python
import awswrangler as wr

s3path = "s3://bucket/prefix/key.parquet"
df = wr.s3.read_parquet(s3path, columns=["time", "Field 1"])
```
Will throw a KeyError. However, the file is retrieved successfully if the `columns` argument is not specified or using `pyarrow`:
```python
import pyarrow.parquet as pq
from s3fs import S3FileSystem

s3 = S3FileSystem()
s3path = "s3://bucket/prefix/key.parquet"

df = pq.ParquetDataset(s3path, filesystem=s3).read(columns=["time", "Field 1"]).to_pandas()
```
Full traceback below:
```python
~/anaconda3/lib/python3.7/site-packages/awswrangler/s3/_read_parquet.py in read_parquet(path, path_suffix, path_ignore_suffix, partition_filter, columns, validate_schema, chunked, dataset, categories, safe, use_threads, last_modified_begin, last_modified_end, boto3_session, s3_additional_kwargs)
    547         return _read_parquet_chunked(paths=paths, chunked=chunked, validate_schema=validate_schema, **args)
    548     if len(paths) == 1:
--> 549         return _read_parquet(path=paths[0], **args)
    550     if validate_schema is True:
    551         _validate_schemas_from_files(

~/anaconda3/lib/python3.7/site-packages/awswrangler/s3/_read_parquet.py in _read_parquet(path, columns, categories, safe, boto3_session, dataset, path_root, s3_additional_kwargs, use_threads)
    376         dataset=dataset,
    377         path=path,
--> 378         path_root=path_root,
    379     )
    380

~/anaconda3/lib/python3.7/site-packages/awswrangler/s3/_read_parquet.py in _arrowtable2df(table, categories, safe, use_threads, dataset, path, path_root)
    232     if metadata:
    233         _logger.debug("metadata: %s", metadata)
--> 234         df = _apply_index(df=df, metadata=metadata)
    235         df = _apply_timezone(df=df, metadata=metadata)
    236     return df

~/anaconda3/lib/python3.7/site-packages/awswrangler/s3/_read_parquet.py in _apply_index(df, metadata)
    173     if index_columns:
    174         if isinstance(index_columns[0], str):
--> 175             df = df.set_index(keys=index_columns, drop=True, inplace=False, verify_integrity=False)
    176         elif isinstance(index_columns[0], dict) and index_columns[0]["kind"] == "range":
    177             col = index_columns[0]

~/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py in set_index(self, keys, drop, append, inplace, verify_integrity)
   4301
   4302         if missing:
-> 4303             raise KeyError(f"None of {missing} are in the columns")
   4304
   4305         if inplace:

KeyError: "None of ['__index_level_0__'] are in the columns"
```

**Comment by igorborgest** [comment]
Hi @tasq-inc, thanks for reporting it.

Do you mind to test it from our development branch and check if it solves your use case?
The ideia is to publish this fix in the version `1.9.5` this weekend.

```sh
pip install git+https://github.com/awslabs/aws-data-wrangler.git@dev
```

**Comment by tasq-inc** [comment]
Yep, this works as intended now. Appreciate the fast turnarounds! You're awesome.

**Comment by igorborgest** [comment]
Released in the version [1.9.5](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.9.5)
